### PR TITLE
Fixed #7891, #3019 and #8260 [sc-15014] - added asset acceptance and signed eula as pdf

### LINKED ISSUE #7891: Feature Request - Store accepted and signed EULA
#### Expected Behavior (or desired behavior if a feature request)

We expect that after an end-user receives and signs the EULA, it can be store in any place into de application, so that we can print it later or , in our case, it also works as evidence for the assigment. Other people have requested this in the past, but those cases have been closed due to lack of follow-up.

-----

#### Actual Behavior

The user receives an email with the asset assigment, he accepts and signs, but that EULA and signature are not store in any place.

-----
#### Provide answers to these questions:

- Is this a fresh install or an upgrade? Upgrade
- Version of Snipe-IT you're running: v4.8.0  build 4186

I just want to say THANKS for making this awesome tool!

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description

Fixes #7891, #3019 and #8260

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Http/Controllers/Account/AcceptanceController.php [hunk 1]
```diff
@@ -118,7 +117,21 @@ public function store(Request $request, $id)
...
+
+    public function SignatureEulaPDF($acceptance, $sig_image)
+    {
+//        When I compact the $data I at least get back an ipsum string in the error message that is recognized as an
+//        "unknown variable". 2nd param in load view should be the array of things you wanted rendered in the pdf.
+        $data = [
+        'eula'=> $acceptance->checkoutable->getEula(),
+        'signature' => $sig_image,
+        ];
+        $pdf = PDF::loadView('account.accept.create', compact($data));
```
Comment (snipe) [inline app/Http/Controllers/Account/AcceptanceController.php]: Why do we need to compact this? It’s already an array,, so we should just be able to pass it?

File: app/Http/Controllers/Api/ReportsController.php [hunk 2]
```diff
@@ -52,6 +52,7 @@ public function index(Request $request)
             'accept_signature',
             'action_type',
             'note',
+            'stored_eula_file',
```
Comment (Godmartinz) [inline app/Http/Controllers/Api/ReportsController.php:55]: I think I need this here ultimately, but I could be wrong.

File: app/Http/Controllers/Account/AcceptanceController.php [hunk 3]
```diff
@@ -119,6 +124,23 @@ public function store(Request $request, $id)
...
+            'logo' => public_path().'/uploads/snipe-logo.png',
+        ];
+
+        \Log::error(storage_path().'/eula-pdfs/'.$sig_filename);
+
+        $pdf = Pdf::loadView('account.accept.accept-eula', $data);
+        $stored_eula = Storage::put('private_uploads/eula-pdfs/accepted-eula-'.date('Y-m-d-h-i-s').'.pdf', $pdf->output());
+
+        //not sure what im doing here,but I think its something of this.
+        route('log.storedeula.download', $stored_eula);
```
Comment (Godmartinz) [inline app/Http/Controllers/Account/AcceptanceController.php]: I want to send the value safely to another file so this is where my brain is starting with this.

File: app/Http/Controllers/Account/AcceptanceController.php [hunk 4]
```diff
@@ -119,6 +128,32 @@ public function store(Request $request, $id)
             $return_msg = trans('admin/users/message.declined');
         }
 
+        $item = $acceptance->checkoutable_type::find($acceptance->checkoutable_id);
+        $assigned_to = User::find($item->assigned_to);
```
Comment (Godmartinz) [inline app/Http/Controllers/Account/AcceptanceController.php]: Will probably  need a switch statement for accessory acceptance. some table headers are different and some columns should also be included. But this covers assets

**Comment by Godmartinz** [comment]
This is completed for now.

**Pull request merged**